edgelet/iotedge-proxy/src/app.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::path::Path;
+
+use clap::{crate_authors, crate_description, crate_name, crate_version, App, Arg};
+use log::info;
+
+use crate::{logging, settings, Error, Settings};
+
+pub fn init() -> Result<Settings, Error> {
+    logging::init();
+
+    info!("Starting proxy server");
+
+    let matches = create_app().get_matches();
+    let config_file = matches
+        .value_of_os("config")
+        .map_or_else(
+            || {
+                let path = Path::new(settings::DEFAULT_SETTINGS_FILEPATH);
+                if path.exists() {
+                    Some(path)
+                } else {
+                    None
+                }
+            },
+            |name| Some(Path::new(name)),
```

Comment:
Should you check if path exists in this as well?

Reply:
It will be checked inside `Settings::new(&path)`. An error will be returned in this case.